web-common/src/features/dashboards/time-dimension-details/tdd-export.ts

```diff
@@ -57,30 +61,26 @@ export function getTDDAggregationRequest(
   )
     return undefined;
 
-  const timeRange = mapTimeRange(
-    timeControlState,
-    dashboardState.selectedTimezone,
-    explore,
-  );
+  let timeRange: V1TimeRange | undefined;
+  if (isScheduled) {
+    timeRange = mapSelectedTimeRangeToV1TimeRange(
+      timeControlState,
+      dashboardState.selectedTimezone,
+      explore,
+    );
+  } else {
+    // NOTE: This is currently needed to ensure the on-demand exports have the same time range as seen on-screen. Currently,
+    // the client-side interpretation of time ranges is not the same as the server-side interpretation.
+    timeRange = {
```

Comment:
This happens because we do a lot of adjusting of time range in UI. The way reports and alerts work is by adding `end` along with iso ranges. We should do the same here for adhoc exports.

```
if (!isScheduled) {
  if (timeRange) {
    timeRange.end = timeControlState.timeEnd;
  }
  if (comparisonTimeRange) {
    comparisonTimeRange.end = timeControlState.comparisonTimeEnd;
  }
}
```